agenta-backend/agenta_backend/routers/variants_router.py

```diff
@@ -331,6 +330,39 @@ async def get_variant(
             app_variant_id=variant_id
         )
 
+        if isCloudEE():
```

Comment:
@aybruhm 
why do we need to pass the app variant to check action access ?
The permission should be enough
If it's not required let's open another PR fixing that and let's fetching app_variant after checking permissons not before

Reply:
> why do we need to pass the app variant to check action access ?

Because that's how it works? We need to pass either the `object` or the `object_id` to the `check_action_action` or `check_rbac_permission` with the permission and check if the user has the permission to do what the permission that was passed wants them to do.

> If it's not required let's open another PR fixing that and let's fetching app_variant after checking permissons not before

That's not the way rbac works. What I did is much easier and reduces duplicate query.